plaidml/core/core.h

```diff
@@ -69,7 +76,14 @@ void call_void(edsl_source_location loc, F fn, Args... args) {
   fn(&err, args...);
   if (err.code) {
     std::stringstream ss;
-    ss << "Exception at " << loc.file_name() << ":" << std::to_string(loc.line()) << " with message: " << str(err.msg);
+    if (!strstr(loc.file_name(), "edsl/edsl.h")) {
+      ss << "Exception at " << loc.file_name() << ":" << std::to_string(loc.line());
+    } else {
+      ss << "Exception at "
+         << "??"
+         << ":" << std::to_string(0);
+    }
+    ss << " with message: " << str(err.msg);
```

Comment:
I see this common pattern of code duplicated. Having a small helper function might be useful.

Reply:
I wonder if we could accomplish this with a custom exception class derived from `std::exception`

Reply:
Sounds good; I'll try out making a custom exception and if I discover some compelling reason not to do that I'll make a helper function.

Reply:
I was getting a bizarre response when I tried to implement the Exception class; everything compiles & runs but the printed error is nonsense (eg "\xC0\x17\x16\x6"). If I put a print the error message in `what()` method right before I return it, it looks fine, but the message doesn't make it through the `return`. I tried several implementations, copying tutorials verbatim, etc; nothing resolved this. Couldn't find any mention of this issue online. So, I used a helper function instead, but if you've seen anything like this before & know how it might be fixed please lmk.

Reply:
The trick with `what()` is that it must return a buffer that lives beyond the lifetime of the method. So you'd need to have a `std::string` member to store the memory. This string would be built up in the constructor of the exception. Then `what()` returns `msg.c_str()`